src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,272 @@
+        } catch (GeePeeTeeException e) {
+            ui.showErrorMessage(e.getMessage());
+        } catch (InvalidTaskIndexException e) {
+            ui.showErrorMessage(e.getMessage());
+        }
+    }
+
+    /**
+     * Processes the event command by attempting to create and add a new event task
+     * to the task list.
+     * @param input The user input to be processed.
+     * @throws InvalidDateException       If the date of the event is invalid.
+     * @throws InvalidTaskFormatException If the input format of the event is
+     *                                    incorrect.
+     * @throws GeePeeTeeException         If an error occurs while adding the event
+     *                                    task.
+     */
+    private void processEventCommand(String input) {
+        try {
+            Event newEvent = Event.createFromInput(input);
+            taskList.addTask(newEvent);
+            ui.showAddTask(newEvent, taskList.getTaskCount());
+            storage.saveTaskList(taskList.getTasksList());
+        } catch (InvalidDateException e) {
+            ui.showErrorMessage(e.getMessage());
+        } catch (InvalidTaskFormatException e) {
+            ui.showErrorMessage(e.getMessage());
+        } catch (GeePeeTeeException e) {
+            ui.showErrorMessage(e.getMessage());
+        }
+    }
+
+    /**
+     * Processes the deadline command by attempting to create and add a new deadline
+     * task to the task list.
+     * @param input The user input to be processed.
+     * @throws InvalidDateException       If the date of the deadline is invalid.
+     * @throws InvalidTaskFormatException If the input format of the deadline is
+     *                                    incorrect.
+     * @throws GeePeeTeeException         If an error occurs while adding the
+     *                                    deadline task.
+     */
+    private void processDeadlineCommand(String input) {
+        try {
+            Deadline newDeadline = Deadline.createFromInput(input);
+            taskList.addTask(newDeadline);
+            ui.showAddTask(newDeadline, taskList.getTaskCount());
+            storage.saveTaskList(taskList.getTasksList());
+        } catch (InvalidDateException e) {
+            ui.showErrorMessage(e.getMessage());
+        } catch (InvalidTaskFormatException e) {
+            ui.showErrorMessage(e.getMessage());
+        } catch (GeePeeTeeException e) {
+            ui.showErrorMessage(e.getMessage());
+        }
+    }
+
+    /**
+     * Processes the todo command by attempting to create and add a new todo task to
+     * the task list.
+     * @param input The user input to be processed.
+     * @throws InvalidTaskFormatException If the input format of the todo is
+     *                                    incorrect.
+     * @throws GeePeeTeeException         If an error occurs while adding the todo
+     *                                    task.
+     */
+    private void processToDoCommand(String input) {
```

Comment:
Maybe divide the Parser class further into Parser and Handler/Processor class to streamline the code and improve readability.